src/ovro_lwa_portal/viz/_data.py

```
@@ -0,0 +1,454 @@
+            out[ti] = slc[l_idx, m_idx]
+        return out
+
+    def spectrum(self, l_idx: int, m_idx: int, time_idx: int) -> np.ndarray:
+        """Get a 1D frequency spectrum at a display pixel and time."""
+        out = np.empty(self.n_freqs, dtype=np.float32)
+        for fi in range(self.n_freqs):
+            slc = self._load_slice(time_idx, fi)
+            out[fi] = slc[l_idx, m_idx]
+        return out
+
+    def nearest_lm_idx(self, l: float, m: float) -> tuple[int, int]:
```

Ruff (0.15.10)

[error] Ambiguous variable name: `l`

(E741)